lib/ruby_smb/dcerpc/winreg.rb

```diff
@@ -124,14 +127,16 @@ def open_key(handle, sub_key)
       end
 
       # Retrieve the data associated with the named value of a specified
-      # registry open key.
+      # registry open key. This will also return the type if required.
       #
       # @param handle [Ndr::NdrContextHandle] the handle for the key
       # @param value_name [String] the name of the value
+      # @param value_name [Boolean] also return the data type if set to true
       # @return [String] the data of the value entry
+      # @return [Array] if `type` is true, an array containing the data type and the actual data of the value entry
```

Comment:
This feels pretty off. Seems like we should have a new `#query_value` method that returns a hash containing the type and data so we could add new fields in the future and keep this one to call the new one and just return the data. This would ensure the return type is consistent rather than being dependent on the arguments the user is supplying.

Do you have any ideas how many places use this method that would need to be refactored if we just changed it directly? If we did that, we could return a struct with `type` and `data` attributes.

Reply:
I agree, it was a quick workaround to solve an issue, but we can think about something more robust.

I quickly checked in RubySMB and Metasploit and `#query_value` is not used that much. I didn't want to make the change directly in this method to avoid breaking changes. This would require a major version bump and might not be what we want.

That said, I'm fine with doing it as long as it is ok to release a new major release of RubySMB.

Otherwise, I also think the other option is good as a temporary change. We can still add this new `#query_value` method, as you suggested, with a warning saying it will disappear in the next major release, once we change the existing `#query_value` method.

Reply:
After our discussion on slack, we decided to go ahead and change the `#query_value` method and return a struct with `type` and `data` attributes. We won't do a major version bump for this, even if it changes slightly the return value type. This method is only used in RubySMB and in one location in Metasploit.